cloudinit/log.py

```diff
@@ -63,6 +63,16 @@ def flushLoggers(root):
     flushLoggers(root.parent)
 
 
+def defineDeprecationLogger(lvl=25):
+    logging.addLevelName(lvl, "DEPRECATED")
```

Comment:
Correct me if I am wrong, but I think we do not refer to this level by the name anywhere.

Taking into account that and https://docs.python.org/3/howto/logging.html#custom-levels I would suggest removing this line.

Reply:
Since this string is printed in the logs, removal of it causes `DEPRECATED` to be replaced with the much less useful `Level 25`, so I lean towards keeping it as is.

If desired I can perhaps add to the documentation a section about using cloud-init as a library, to communicate to users logging expectations.

Reply:
@aciba90 I just updated this PR with updated docs per this comment, as well as some extra tests. Requesting re-review.

Reply:
We talked it over a bit and I'm ok with these changes. Since cloud-init isn't _really_ used as a library in too many places, I don't think we need to worry about conflicts.